File: frontAnimalCrossing/src/app/models/island.ts
import { IslandLeader } from "./islandLeader";

export class Island {



  constructor(
    private _name: string,
    private _idIslandLeader: number,
    private _villagers: string[]

  ) { }
  public get villagers(): string[] {
    return this._villagers;
  }
  public set villagers(value: string[]) {
    this._villagers = value;
  }
  public get idIslandLeader(): number {
    return this._idIslandLeader;
  }
  public set idIslandLeader(value: number) {
    this._idIslandLeader = value;
  }

  public get name(): string {
    return this._name;
  }
  public set name(value: string) {
    this._name = value;
  }


  public toJSON(): any{
    return {
      name: this._name,
      idIslandLeader: this._idIslandLeader,
      villagers: this._villagers

    }
  }
}

File: backAnimalCrossing/src/main/java/com/ironhack/backAnimalCrossing/model/Island.java
package com.ironhack.backAnimalCrossing.model;

import javax.persistence.*;

@Entity
public class Island {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    @OneToOne
    @JoinColumn(name = "islandLeader_id")
    private IslandLeader islandLeader;


    private String[] villagers;

    public Island() {
    }

    public Island(String name, IslandLeader islandLeader, String[] villagers) {
        this.name = name;
        this.islandLeader = islandLeader;
        this.villagers = villagers;
    }

    public Island(String name) {
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public IslandLeader getIslandLeader() {
        return islandLeader;
    }

    public void setIslandLeader(IslandLeader islandLeader) {
        this.islandLeader = islandLeader;
    }

    public String[] getVillagers() {
        return villagers;
    }

    public void setVillagers(String[] villagers) {
        this.villagers = villagers;
    }
}
